src/main/java/com/github/bordertech/config/DefaultConfiguration.java

```diff
@@ -1543,11 +1366,15 @@ protected void handlePropertiesChanged() {
 		subcontextCache.clear();
 		// Check if environment changed
 		checkEnvironmentProperty();
+		setEnvironmentProfile();
 	}
 
 	/**
 	 * Check if the environment property has been set.
+	 *
+	 * @deprecated - Replaced by Environment Profile
 	 */
+	@Deprecated
 	protected void checkEnvironmentProperty() {
```

Comment:
I hope this makes sense... can we provide new methods using `profile` like `checkProfileProperty`, `useProfileKey`, `getProfileKey` and the old deprecated `checkEnvironment` etc methods just call the new profile methods.

Reply:
Looks good.